### PR TITLE
Accumulate

### PR DESCRIPTION
**Pull request by jonenst**
Hi,
PR for #1547 
What do you think ?

### CODE REVIEW

**Comment by jonenst** [comment]
Hmm wait let me add a note saying that this is know as scan in other languages, I'll push -f to avoid a tiny commit

**Comment by jonenst** [comment]
Ok. I haven't bootstrapped with this change yet (takes ~30 minutes on my machine)

File: extra/benchmark/hashtables/hashtables.factor [hunk 1]
```diff
@@ -7,7 +7,7 @@ QUALIFIED: assocs
 IN: benchmark.hashtables
 
 MEMO: strings ( -- str )
-    1 100 [a,b] 1 [ + ] accumulate nip [ number>string ] map ;
+    1 100 [a,b] 0 [ + ] accumulate* [ number>string ] map ;
```
Comment (mrjbq7) [inline extra/benchmark/hashtables/hashtables.factor]: These aren't the same thing, though?  (I know it's just a benchmark, but...)

Comment (jonenst) [inline extra/benchmark/hashtables/hashtables.factor]: I think it's the same?
Le 8 mars 2016 15:41, "John Benediktsson" <notifications@github.com> a
écrit :

Comment (mrjbq7) [inline extra/benchmark/hashtables/hashtables.factor]:
```factor
IN: scratchpad 1 100 [a,b] 1 [ + ] accumulate nip 10 head .
{ 1 2 4 7 11 16 22 29 37 46 }

IN: scratchpad 1 100 [a,b] 0 [ + ] cum-map 10 head .
{ 1 3 6 10 15 21 28 36 45 55 }
```

Comment (mrjbq7) [inline extra/benchmark/hashtables/hashtables.factor]: This is however, the same thing:

```factor
IN: scratchpad 0 100 [a,b) 1 [ + ] cum-map 10 head .
{ 1 2 4 7 11 16 22 29 37 46 }
```

Comment (jonenst) [inline extra/benchmark/hashtables/hashtables.factor]: Oops, yes you're right. Thanks for catching it. If it doesn't change the
benchmark, maybe we can switch it to the more classical 1 3 6 10 [...]
sequence in another commit?

In extra/benchmark/hashtables/hashtables.factor
<https://github.com/factor/factor/pull/1548#discussion_r55370370>:

> @@ -7,7 +7,7 @@ QUALIFIED: assocs
>  IN: benchmark.hashtables
>
>  MEMO: strings ( -- str )
> -    1 100 [a,b] 1 [ + ] accumulate nip [ number>string ] map ;
> +    1 100 [a,b] 0 [ + ] accumulate* [ number>string ] map ;

This is however, the same thing:

IN: scratchpad 0 100 [a,b) 1 [ + ] cum-map 10 head .
{ 1 2 4 7 11 16 22 29 37 46 }

—
Reply to this email directly or view it on GitHub
<https://github.com/factor/factor/pull/1548/files#r55370370>.

**Comment by mrjbq7** [comment]
Are you going to fix the benchmark thing and re-push?  Then I can merge!

**Comment by jonenst** [comment]
Yup!
Le 8 mars 2016 16:55, "John Benediktsson" <notifications@github.com> a
écrit :

**Comment by jonenst** [comment]
Rebased on master and "make-my-image" then bootstrap works fine

**Comment by jonenst** [comment]
Also changing the strings of the hashtables benchmark changed the performance so I used your expression to leave the old sequence

**Comment by mrjbq7** [comment]
Merged, thanks!!

**Pull request closed without merging**